### PR TITLE
Add 'Update Details' Button in Community Nurse Log Update Details Page

### LINKED ISSUE #9060: Add "Update Details" Button in view details page of community nurse and details log update
**Describe the bug**

We can't directly access the edit log update page from the view details page of the community nurse or the detail log update page.

**To Reproduce**

1. Go to the patient tab
2. Click on active consultation and click on log update
3. create a new community nurse log update
4. On successful creation, in the patient consultation page, switch to "Daily Rounds" view
5. click the view details page in the community nurse log update card
6. See error

**Expected behavior**

- [ ] add the "Update details" button, in the view details page of community and detail log update

**Screenshots**

Image

Image

### PR DESCRIPTION
**Pull request by kihan2518B**
## Proposed Changes

- Fixes #9060 

@ohcnetwork/care-fe-code-reviewers

## Merge Checklist

- [ ] Add specs that demonstrate bug / test a new feature.
- [ ] Update [product documentation](https://docs.ohc.network).
- [ ] Ensure that UI text is kept in I18n files.
- [ ] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [ ] Request for Peer Reviews
- [ ] Completion of QA

## Summary by CodeRabbit

- **New Features**
	- Added an "Update Log" button in the Critical Care Preview component for easy access to update daily rounds details.
	- Introduced a new `primary` button variant with enhanced styles for light and dark themes, expanding visual options for users.
	- Updated the Daily Round List Details component to include a new button with a CareIcon for improved user interaction.
- **Bug Fixes**
	- Corrected the button label in the PatientLogupdate class from "Update Details" to "Update Log" for consistency.

### CODE REVIEW

File: src/components/LogUpdate/CriticalCarePreview.tsx [hunk 1]
```diff
@@ -57,6 +57,15 @@ export default function CriticalCarePreview(props: Props) {
...
         </ButtonV2>
+        <div>
+          <div className="mt-2">
+            <ButtonV2
+              href={`/facility/${props.facilityId}/patient/${props.patientId}/consultation/${props.consultationId}/daily-rounds/${props.id}/update`}
+            >
+              Update Details
+            </ButtonV2>
+          </div>
+        </div>
```
[the remark this answers is not in the document]
Comment (kihan2518B) [reply src/components/LogUpdate/CriticalCarePreview.tsx]: Screenshot from 2024-11-13 21-48-39
This might looks like this

**Comment by kihan2518B** [comment]
@rithviknishad can you review my pr

**Comment by nihal467** [comment]
image
image
@kihan2518B 
- [ ] As per the existing design of "Brief Update" and the issue description, position the "Update Details" button at the top-right corner of the card for both the "Community Nurse" and "Detail Log Update" view details pages.

File: src/components/LogUpdate/CriticalCarePreview.tsx [hunk 2]
```diff
@@ -49,14 +49,26 @@ export default function CriticalCarePreview(props: Props) {
 
   return (
     <div className="w-full transition-all duration-200 ease-in-out md:mx-auto md:max-w-5xl md:pt-8">
-      <div className="py-4">
+      <div className="py-4 flex items-center justify-between max-[368px]:flex-col max-[368px]:items-start">
```
Comment (rithviknishad) [inline src/components/LogUpdate/CriticalCarePreview.tsx]: if it's for mobile view use `md:` modifier instead

```suggestion
      <div className="py-4 flex items-center justify-between md:flex-col md:items-start">
```

File: src/components/LogUpdate/CriticalCarePreview.tsx [hunk 3]
```diff
@@ -49,14 +49,26 @@ export default function CriticalCarePreview(props: Props) {
...
+            <ButtonV2
+              id="update-log"
+              data-testid="update-log-button"
+              aria-label={t("update_log")}
+              href={`/facility/${props.facilityId}/patient/${props.patientId}/consultation/${props.consultationId}/daily-rounds/${props.id}/update`}
+            >
+              {t("update_log")}
+            </ButtonV2>
+          </div>
+        </div>
```
Comment (rithviknishad) [inline src/components/LogUpdate/CriticalCarePreview.tsx]: why wrap in two divs? you can directly set the `"mt-2"` class in the Button itself.

Also let's switch to using the shadcn's button instead while we are at it. We are using that in all new UIs.

**Review by rithviknishad — CHANGES_REQUESTED** [review]

**Comment by nihal467** [comment]
image

@kihan2518B keep the button colour primary ,

File: src/components/LogUpdate/CriticalCarePreview.tsx [hunk 4]
```diff
@@ -49,14 +52,27 @@ export default function CriticalCarePreview(props: Props) {
...
         >
           {t("back_to_consultation")}
         </ButtonV2>
+        <Link
+          className="mt-2"
+          href={`/facility/${props.facilityId}/patient/${props.patientId}/consultation/${props.consultationId}/log_updates/${props.id}/update`}
+        >
+          <Button
+            id="update-log"
+            className="bg-primary text-white"
```
Comment (rithviknishad) [inline src/components/LogUpdate/CriticalCarePreview.tsx]: set the variant prop to primary instead

**Review by rithviknishad — CHANGES_REQUESTED** [review]

File: src/components/LogUpdate/CriticalCarePreview.tsx (hunk 4 above)
Comment (kihan2518B) [reply src/components/LogUpdate/CriticalCarePreview.tsx]: image

Outline_Primary?

Comment (rithviknishad) [reply src/components/LogUpdate/CriticalCarePreview.tsx]: You can either use `outline_primary`, or copy the `primary` implementation from: https://github.com/ohcnetwork/care_fe/pull/9076/files#diff-c29ec4c06502c45a9ee8740156618a5823f7cadbcd01a339304a66537c0dc30dR14-R15

File: src/components/LogUpdate/CriticalCarePreview.tsx (hunk 2 above)
Comment (rithviknishad) [reply src/components/LogUpdate/CriticalCarePreview.tsx]: @kihan2518B This is still unresolved btw

File: src/components/LogUpdate/CriticalCarePreview.tsx [hunk 5]
```diff
@@ -49,14 +52,27 @@ export default function CriticalCarePreview(props: Props) {
...
+        >
+          <Button
+            id="update-log"
+            variant={"primary"}
+            data-testid="update-log-button"
+            aria-label={t("update_log")}
+          >
+            {t("update_log")}
+          </Button>
+        </Link>
```
Comment (rithviknishad) [inline src/components/LogUpdate/CriticalCarePreview.tsx]: This right way to compose links using button is as shown.

Eg. usage in care: https://github.com/ohcnetwork/care_fe/blob/532945766d2b76bd64dc4ce7ced9aceedece313d/src/components/Common/Breadcrumbs.tsx#L77-L83

See shadcn's docs: https://ui.shadcn.com/docs/components/button#as-child

**Review by rithviknishad — CHANGES_REQUESTED** [review]
Do replace existing ButtonV2's with shadcn's button too btw. Also previously requested changes are still not resolved.

**Comment by rithviknishad** [comment]
Kindly request for review only once all changes requested are resolved.

File: src/components/LogUpdate/CriticalCarePreview.tsx (hunk 2 above)
Comment (kihan2518B) [reply src/components/LogUpdate/CriticalCarePreview.tsx]: Got this in last commit.

**Comment by rithviknishad** [comment]
@coderabbitai review

File: src/components/LogUpdate/CriticalCarePreview.tsx [hunk 6]
```diff
@@ -49,14 +52,28 @@ export default function CriticalCarePreview(props: Props) {
 
   return (
     <div className="w-full transition-all duration-200 ease-in-out md:mx-auto md:max-w-5xl md:pt-8">
-      <div className="py-4">
+      <div className="py-4 flex items-center justify-between md:[368px]:flex-col md:items-start">
```
Comment (rithviknishad) [inline src/components/LogUpdate/CriticalCarePreview.tsx]: What does `md:[368px]:flex-col` imply?

Comment (kihan2518B) [reply src/components/LogUpdate/CriticalCarePreview.tsx]: Ahh it was typo mistake, Sorry :(

**Review by rithviknishad — APPROVED** [review]

**Review by Jacobjeevan — APPROVED** [review]

**Comment by nihal467** [comment]
image

- [x] the button colour primary is different from the update details button in the brief
- [x] there is no hovering effect for the button

@kihan2518B Ensure that the color and effects of the button you created remain consistent with the existing ones, as the requirement was to implement a button similar to the one already present on the platform.

**Comment by rithviknishad** [comment]
@nihal467 we are switching to newer components (shadcn buttons) in newer implementations right.

File: src/components/LogUpdate/CriticalCarePreview.tsx (hunk 4 above)
Comment (kihan2518B) [reply src/components/LogUpdate/CriticalCarePreview.tsx]: > set the variant prop to primary instead

@nihal467  rithvik asked me to do so

**Comment by kihan2518B** [comment]
hey @nihal467 what should I do now?

**Review by rithviknishad — APPROVED** [review]

**Comment by kihan2518B** [comment]
@rithviknishad what am I supposed to do here?

**Comment by rithviknishad** [comment]
It's queued for testing @kihan2518B

**Comment by nihal467** [comment]
LGTM

**Pull request merged**